fix(cgroups): PodMetadata.IPs as low level net.IP

nerzhul opened the pull request:
We are currently working on finding what happen on #33063, and during the debug logging analysis and the code reading i found that pod metadata fetch on our very busy clusters is done very very very often, hundreds of time per second, because it's done per socket creation.

Each time we retrieve those metadata we currently parse string adresses we copy over the metadata store to the caller through channels and we parse it on each metadata call.

With this change, we parse it on the store directly, once and we can directly play with it on the callers without converting each time we call it. This will reduce CPU cycles each time we create a socket, and also the memory of the metadata store because we use a raw 4 bytes instead of 15 bytes (for IPv4) string

```release-note
Reduce memory footprint for cgroup PodMetadata tracing & CPU usage on each socket creation.
```

note: it's totally backportable on 1.15 branch

aditighag reviewed (approved):
Thanks! Jfyi, we recently optimized the related code path to cache pod metadata so that channel invocations are reduced - https://github.com/cilium/cilium/pull/32615.

nerzhul committed:
fix(cgroups): PodMetadata.IPs as low level net.IP

We are currently working on finding what happen on #33063, and during the debug logging analysis
and the code reading i found that pod metadata fetch on our busy clusters is done often,
hundreds of time per second, as it's done per socket creation.

Each time we retrieve those metadata, we currently copy from the metadata store to the caller
through channels and it's parsed on each metadata call.

With this change, we parse it on the store directly, once and we can play with it on the
callers without converting each time we need it.

This will reduce CPU cycles each time we create a socket, and also the memory of the metadata
store from 15 bytes to 4 bytes (per IPv4)

Signed-off-by: Loic Blot <loic.blot@unix-experience.fr>

ldelossa reviewed (approved):
LGTM :+1:

squeed commented on pkg/cgroups/manager/manager.go:
```diff
@@ -75,15 +76,15 @@ type cgroupManager struct {
 	Name      string
 	Namespace string
-	IPs       []string
+	IPs       []net.IP
```
Just adding a comment that this should be a `netip.Addr`. That way we won't accidentally merge.
